### ISSUE TITLE
Feature request - Dump the character JSON sheet out with each new chat (for versioning control).

### ISSUE THREAD

**Issue by erew123** [TavernAI/TavernAI-v1#221]
Sometimes I edit a character multiple times over a long period, which alters how the character behaves vastly. Sometimes for the worse and you want to revert back to how a character was in a previous chat, maybe one from a month ago where you've changed the character multiple times since (yes even using the same settings and model you used back then).

As such I'm proposing dumping out the character sheet at the top of the logged chats, so you can always go back find the setup you had for the character, re-create that character sheet anew or maybe figure what it was you had setup back then, that made the character work better.

**OR**

**Maybe a simple solution** would be to dump the character JSON into another file alongside the logged chat e.g. 

...\public\chats\charactername\1690472902128.jsonl **(the chat log)**
...\public\chats\charactername\1690472902128-CHAR.jsonl **(the character json used in the matching chat. Dump it out at the start of a chat new log file being created and no need to re-create it or change it again after that point)**

I appreciate its extra files, but it does give the user some kind of simple versioning control where they don't have to constantly back up their characters each time they decide to try something new.

Thanks

**Comment by TavernAI** [comment]
It's not very clear at what exact moment it needs to take a snapshot of char. At the moment when the chat was created? But as far as I know characters are often customized for a particular chat after creating the chat.

**Comment by erew123** [comment]
Yes, I think the character would only need a snapshot made when the chat is created e.g. "Start New Chat" is selected, creating a matching character sheet/json backup for THAT chat log, so you can back reference any previous chat log and see how you had the character setup for THAT exact chat.

What I was getting at with my suggestion is that, for example, I may re-use a chat character for say 20+ chats, but after a few of those chats, change one or two things to try something out or make the AI pick up on something a bit stronger. And then maybe a few chats down the line, more changes, and so on and so forth. By the time you get to the 20th chat or more, some things may be acting differently and you're trying to remember what it was you had that worked 20 chats ago..... but there's no easy way to look back at the character sheet/Json, unless you back them up each time you make a change.

Hope that makes some sort of sense.

Thanks

**Comment by yggdrasil75** [comment]
agreed. though maybe the best option would be that when creating a new chat, ask "do you want to customize the characters" and create a "clone" of the character in a subfolder that will be edited when you edit the character within that specific chat.
